packages/x-tree-view/src/TreeItem2/TreeItem2.tsx

```diff
@@ -155,6 +167,48 @@ export const TreeItem2GroupTransition = styled(Collapse, {
   ],
 });
 
+function UnstyledLabelInput({ onChange, value, ...props }: UseTreeItem2LabelInputSlotOwnProps) {
+  const [labelInputValue, setLabelInputValue] = React.useState(value);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    onChange?.(event);
+    setLabelInputValue(event.target.value);
+  };
+
+  const resetLabel = React.useCallback(() => setLabelInputValue(value), [value]);
+
+  React.useEffect(() => {
+    return resetLabel();
+  }, [resetLabel]);
+
+  return (
+    <input
+      {...props}
+      onChange={handleInputChange}
+      value={labelInputValue as string}
```

Comment:
Why do we need to cast here?
If we really need to cast, I would do it at the `useState` level, but isn't `value` a `string` in the typing?

Reply:
> but isn't value a string in the typing

It's a reactNode and it's optional 🤔 this would work :
![image](https://github.com/user-attachments/assets/3a27c581-debd-4ed0-9b68-c4512f030b41)

or we could change the typing of the label in TreeItem2?

Reply:
You can cast it like you are doing on your screen, since on `RichTreeView` `label` is **always** a string

Reply:
And I would correctly handle the nullable aspect by doing `label ?? ''`

Reply:
I think this is the last ongoing conversation on my side :+1: